src/big_core_base.cpp

```diff
@@ -1,120 +1,99 @@
-// #include <cstdint> RV: smazat, toto je nutne uz v hlavicce
-
 #include "../include/big_core_base.h"
 
-// RV: general question. Proc vsude pouzivas "this->..."? Osobne mi to prijde mene prehledne a nepouzival bych to. Ale predpokladam, ze pro Tebe to bude naopak. Nebo je jiny duvod?
-// RV: general comment. Pri pouziti struktury "if (condition) statement" bud davej vse na jeden radek nebo prosim pouzivej { }. Pokud je to na vice radku bez slozenych zavorek, delam pak nekdy chybu, ze pripisu dalsi statement da dalsi radek a zapomenu pridat { }. Diky
-// RV: general comment. Vsechny komentare, ktere pravdepodobne nebudou soucasti finalni verze a jsou to spise otazky nebo odpovedi na me komentare prosim oznacuj inicialy LB, podobne jako to delam ja. Takoveto komentare mohou byt cesky bez diakritiky.
-// RV: general comment. V cpp souborech dodrzuj prosim pokud mozno stejne poradi metod jako v hlavickovem souboru. Lepe se pak hleda a kontroluje, co je naimplementovano.
-
-BigCoreBase::BigCoreBase()
-    // RV: nasledujici dva radky bych smazal, nevim proc tam cokoli nastavovat, dokud nemam parametry (rozmery, datove typy, atd.)
-	//:
-	//outermostEntitiesOffsets(0)
-{}
-
 BigCoreBase::~BigCoreBase()
 {
-    if (this->data) delete[] this->data;
-}
-
-uint64_t BigCoreBase::getNumberOfImages()
-{
-	return this->numberOfImages;
-}
-
-uint64_t BigCoreBase::getNumberOfTiles()
-{
-	return this->numberOfTiles;
-}
-
-uint64_t BigCoreBase::getImageHeight()
-{
-	return this->imageHeight;
-}
-
-uint64_t BigCoreBase::getImageWidth()
-{
-	return this->imageWidth;
-}
-
-uint64_t BigCoreBase::getNumberOfPlanes()
-{
-	return this->numberOfPlanes;
-}
-
-const std::vector<uint64_t>& BigCoreBase::getDataOrder()
-{
-    return dataOrder;
-}
-
-const std::vector<uint8_t>& BigCoreBase::getDataType()
-{
-    return dataType;
-}
-
-bool BigCoreBase::isEmpty()
-{
-	return this->dataLength == 0 ? true : false;
+    delete[] _data;
 }
 
 void BigCoreBase::clear()
 {
-	if (this->data) delete[] this->data;
+    numberOfImages = 0;
+    imageHeight = 0;
+    imageWidth = 0;
+    numberOfPlanes = 1;
+    dataOrder = { DataOrderIds::NUMBER_OF_IMAGES, DataOrderIds::IMAGE_HEIGHT, DataOrderIds::IMAGE_WIDTH, DataOrderIds::NUMBER_OF_PLANES };
+    dataType = { DataTypes::FLOAT };
+    delete[] _data;
```

Comment:
stejny pripad jako u toho destruktoru

Reply:
Podle specifikace to může být nullptr: If this is a null-pointer, the function does nothing. Podmínka je tedy zbytečná.